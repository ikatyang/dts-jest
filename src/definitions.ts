import * as ts from 'typescript';

export type TriggerDescription = null | string;

export interface ITrigger {
  line: number;
  kind: TriggerKind;
  description: TriggerDescription;
}

export interface ITriggerDescriptions {
  [line: number]: TriggerDescription;
}

export enum TriggerKind {
  None,
  Skip,
  Only,
  Show,
}

export interface ITSConfig {
  extends?: string;
  compilerOptions?: ts.CompilerOptions;
}

export type IRawTSConfig = undefined | string | ts.CompilerOptions;

export interface ISelfConfig {
  tsconfig?: IRawTSConfig;
  reporter?: string;
}

export const default_reporter_template = '\nInferred\n\n{{expression,2}}\n\nto be\n\n{{snapshot,2}}';

export interface IJestConfig {
  globals: {
    _dts_jest_: ISelfConfig;
  };
}

export const config_namespace = '_dts_jest_';
export const runtime_namespace = '_dts_jest_runtime_';

export const trigger_regex = /^\s*\/\/\s*@dts-jest\b(:\S*)?\s*(.+)?\s*$/;
export enum TriggerMatchIndex {
  Input,
  Flag,
  Description,
}
export type ITriggerMatchArray = [
  string,
  string | undefined,
  string | undefined
];

export interface IRuntimeSnapshotParameters {
  line: number;
}
export interface IRuntimeDescriptionParameters {
  line: number;
}
export interface IRuntimeReportParameters {
  line: number;
}

export interface IRuntimeMethods {
  report(parameters: IRuntimeReportParameters): string;
  snapshot(parameters: IRuntimeSnapshotParameters): string;
  description(parameters: IRuntimeDescriptionParameters): string;
}
