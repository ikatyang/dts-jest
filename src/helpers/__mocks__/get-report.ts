export const get_report = () => '[report]';
