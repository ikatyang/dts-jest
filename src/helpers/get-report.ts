import {indent} from '../utils/indent';

// tslint:disable:no-magic-numbers

export const get_report = (reporter_template: string, expression: string, snapshot: string) =>
  reporter_template
    .split(/\{\{(expression|snapshot),([0-9]+)\}\}/g)
    .map((chunk, index, chunks) =>
      (index % 3 === 0)
          ? chunk
          : (index % 3 === 1)
            ? ''
            : (chunks[index - 1] === 'snapshot')
              ? indent(snapshot, +chunk)
              : indent(expression, +chunk),
    )
    .join('');
