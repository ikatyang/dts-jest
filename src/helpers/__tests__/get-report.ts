jest.unmock('../get-report');

import {default_reporter_template} from '../../definitions';
import {get_report} from '../get-report';

const expression = '[expression]';
const snapshot = '[snapshot]';

it('should return correctly', () => {
  expect(get_report(default_reporter_template, expression, snapshot)).toMatchSnapshot();
});
