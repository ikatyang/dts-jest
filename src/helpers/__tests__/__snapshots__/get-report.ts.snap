// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correctly 1`] = `
"
Inferred

  [expression]

to be

  [snapshot]"
`;
